### PR TITLE
Mobile splitpage

### PR DESCRIPTION
**Pull request by swanka**

### CODE REVIEW

File: packages/ui/src/TextField.tsx [hunk 1]
```diff
@@ -279,8 +279,6 @@ export function TextField({
                 width: "100%",
                 color: Unifier.theme.darkGray,
                 fontFamily: Unifier.theme.primaryFont,
-                // Remove border in web.
```
Comment (joshgachnang) [inline packages/ui/src/TextField.tsx:282]: Unrelated?

File: packages/ui-demo/src/SplitPage.stories.tsx [hunk 2]
```diff
@@ -100,32 +97,54 @@ const ManyChildren = () => {
...
 };
 
+const ScrollableContent = () => {
+  const items = Array.from(Array(100).keys()).map((i) => ({
+    name: `user${i}`,
+    id: i,
+  }));
+
+  return (
+    // <FlatList
```
Comment (joshgachnang) [inline packages/ui-demo/src/SplitPage.stories.tsx:127]: Leftover

**Review by joshgachnang — COMMENTED** [review]
Nice, this code is really clean and readable, great job! I did notice on the two children, the first child still gets rendered to the side? 
image

File: packages/ui/src/TextField.tsx (hunk 1 above)
Comment (swanka) [reply packages/ui/src/TextField.tsx:282]: This is causing the mobile apps to crash due to this property being deprecated. I should have placed a TODO on here to add the styles to mimic this behavior.

**Comment by swanka** [comment]
> Nice, this code is really clean and readable, great job! I did notice on the two children and many children stories, the first child still gets rendered to the side? image

I _believe_ this is because you're using the deploy preview which is only a preview of the mobile browser. It seems like the <component>.native.tsx doesn't get triggered if that's the case. You'll probably need to pull down the branch and test through expo. I wonder if there's a way to deploy the mobile app versions for each PR as well

**Comment by swanka** [comment]
Oh I just realized that there's a preview on mobile QR code but that also brings up the browser in mobile.

**Comment by joshgachnang** [comment]
Right, .native.tsx is only going to be on iOS/Android, but it should still render correctly for web right?

**Comment by joshgachnang** [comment]
And I'd love to hook demo up to EAS or something so we could do previews on mobile like we do with our app! Just gotta find the time

File: packages/ui/src/TextField.tsx (hunk 1 above)
Comment (joshgachnang) [reply packages/ui/src/TextField.tsx:282]: Oh interesting, and weird!

**Comment by joshgachnang** [comment]
I think the breakpoints need adjusting. iPhone 14 Max shows 2 side by side columns for the many children. But otherwise looks good on iOS (minus the crash, which might be transient or weird local issues if you're not having an issue on yours)

File: packages/ui-demo/src/SplitPage.stories.tsx [hunk 3]
```diff
@@ -124,18 +124,20 @@ const ScrollableContent = () => {
...
-      keyExtractor={(item) => item.name}
-      renderItem={(item) => {
-        return (
-          <Box key={item.item.name} color="purple" padding={2}>
-            <Text>name: {item.item.name}</Text>
-          </Box>
-        );
-      }}
-    />
+    <ScrollView>
```
Comment (joshgachnang) [inline packages/ui-demo/src/SplitPage.stories.tsx:127]: I'm confused what this is testing, with a FlatList inside a full height scrollview. You generally would want to avoid doing that. You could do one child as a flat list and one as a scrollview with a bunch of content as a good test though (in the multiple children one?)

File: packages/ui/src/SplitPage.native.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
+import React, {Children, useCallback, useEffect, useState} from "react";
```
Comment (joshgachnang) [inline packages/ui/src/SplitPage.native.tsx:1]: One thing where splitting this native vs web like this will bite us is when we try to support apps on tablets. Those would want to get the other experience, but would get a giant phone experience right now.

Not needed for this PR, but something we'll likely want to tackle in the medium term.

**Review by joshgachnang — APPROVED** [review]
Wooooooot! This works so well!

File: packages/ui-demo/src/SplitPage.stories.tsx (hunk 3 above)
Comment (swanka) [reply packages/ui-demo/src/SplitPage.stories.tsx:127]: Yea so wrapping the flat list in the scroll view is actually how I get the scroll to even work. If I remove it, the scroll stops working. I think it has to do with a flatlist/scrollview within the swiper library. The swiper library is actually built off of flatlist. The reason this is a test is because the content in flourish is all scrollable with the exception of chat view. I just want to make sure that children are still scrollable but I don't want to add another embedded scroll view when rendering the child if it's not required. The parent component can add a scrollview if the child isn't scrolling as intended. 

I hope that makes sense but definitely down to talk this through with you if that came out super confusing lol. Especially since you might have a cleaner solution

Comment (joshgachnang) [reply packages/ui-demo/src/SplitPage.stories.tsx:127]: That seems very odd, but hey, it works! I'm ok merging as-is and we can revisit later or if it becomes an issue.
